spec/unit/provider/postgresql_conf/ruby_spec.rb

```diff
@@ -30,6 +30,13 @@
     expect(provider).to respond_to(:exists?)
   end
 
+  describe 'duplicate keys in postgresql.conf' do
+    it 'raises an exception' do
+      allow(provider).to receive(:parse_config).and_return([{ key: 'foo', line: 1 }, { key: 'foo', line: 2 }])
+      expect(provider.exists?).to raise_error(Puppet::Error, 'found multiple config items of foo found, please fix this')
```

Comment:
I think you need to use a block:
```suggestion
      expect { provider.exists? }.to raise_error(Puppet::Error, 'found multiple config items of foo found, please fix this')
```